### PR TITLE
check axis range for LayerNorm

### PR DESCRIPTION
**Pull request by yufenglee**
### Description

Add check on axis to make sure it is in a valid range

### CODE REVIEW

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 1]
```diff
@@ -24,6 +24,17 @@
 #pragma warning(disable : 26426)
 #endif
 namespace ONNX_NAMESPACE {
+
+inline int64_t HandleNegativeAxis(int64_t axis, int64_t rank) {
+  if (axis >= rank || axis < -rank) {
+    fail_shape_inference("axis ", axis,
+                         " is not in valid range [-", rank, ",", rank - 1, "]");
```
Comment (snnn) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:31]: So the range is not symmetric?

Comment (snnn) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:31]: Is rank non-negative? Was it checked anywhere?

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 2]
```diff
@@ -24,6 +24,17 @@
...
+  }
+
+  // Handle negative axis
+  return axis < 0 ? axis + rank : axis;
```
Comment (snnn) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:35]: So "axis + rank" still could overflow?

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc (hunk 1 above)
Comment (yufenglee) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:31]: the range is [-rank, rank-1]

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc (hunk 2 above)
Comment (yufenglee) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:35]: axis is in [-rank, -1], then axis + rank is in [0, rank-1]

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc (hunk 1 above)
Comment (yufenglee) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:31]: rank takes in the size of dim.

**Review by pranavsharma — COMMENTED** [review]
Can you add a test case?

**Review by pranavsharma — COMMENTED** [review]
Also, please fill out the PR template to the best of your ability.

**Review by pranavsharma — COMMENTED** [review]
There might be other places that require a similar fix. For e.g. https://github.com/microsoft/onnxruntime/blob/8f34c8c8eda2890b502021283c463d0552e45b8d/onnxruntime/core/graph/contrib_ops/contrib_defs.cc#L1508-L1511. Can we fix them too?

**Comment by yufenglee** [comment]
> 

This one already has check and its check is different with LayerNorm

**Review by snnn — APPROVED** [review]

**Pull request merged**